File: DotNet/Kyu6/FindTheUniqueNumber.cs
// https://www.codewars.com/kata/585d7d5adb20cf33cb000235/train/csharp

using System;
using System.Collections.Generic;

namespace DotNet.Kyu6;

public class FindTheUniqueNumber
{
    public static int GetUnique(IEnumerable<int> numbers)
    {
        var counts = new Dictionary<int, int>();
        foreach (var number in numbers)
        {
            if (!counts.TryAdd(number, 1)) { counts[number]++; }
        }

        foreach (var count in counts)
        {
            if (count.Value == 1)
            {
                return count.Key;
            }
        }

        return 0;
    }

    public static void Main()
    {
        Console.WriteLine($"{GetUnique([1, 2, 2, 2])} == 1");
    }
}

File: Python/Kyu6/find-the-unique-number.py
# https://www.codewars.com/kata/find-the-unique-number-1/train/python

from unittest import TestCase

def find_uniq(arr):
    return list(filter(lambda x: arr.count(x) == 1, set(arr)))[0] or 0

test = TestCase()
test.assertEqual(find_uniq([ 1, 1, 1, 2, 1, 1 ]), 2)
test.assertEqual(find_uniq([ 0, 0, 0.55, 0, 0 ]), 0.55)
test.assertEqual(find_uniq([ 3, 10, 3, 3, 3 ]), 10)